test/trento/users_test.exs

```diff
@@ -0,0 +1,216 @@
+                 email: user2.email
+               })
+
+      assert changeset.errors[:email] ==
+               {"has already been taken",
+                [{:constraint, :unique}, {:constraint_name, "users_email_index"}]}
+    end
+
+    test "update_user/2 does not update deleted_at" do
+      user = create_user()
+      {:ok, user} = Users.get_user(user.id)
+      assert {:ok, %User{} = user} = Users.update_user(user, %{deleted_at: DateTime.utc_now()})
+      assert user.deleted_at == nil
+    end
+
+    test "update_user/2 lock the user if enable false is passed as attribute and the user is not locked" do
+      user = create_user()
+      {:ok, user} = Users.get_user(user.id)
+      assert {:ok, %User{} = user} = Users.update_user(user, %{enabled: false})
+      refute user.locked_at == nil
+    end
+
+    test "update_user/2 does no lock the user if enable true is passed as attribute and the user is not locked" do
+      user = create_user()
+      {:ok, user} = Users.get_user(user.id)
+      assert {:ok, %User{} = user} = Users.update_user(user, %{enabled: true})
+      assert user.locked_at == nil
+    end
+
+    test "update_user/2 does not relock the user if enable false is passed as attribute and the user is already locked" do
+      user = create_user()
+      {:ok, user} = Users.get_user(user.id)
+
+      assert {:ok, %User{locked_at: locked_at} = user} =
+               Users.update_user(user, %{enabled: false})
+
+      assert {:ok, %User{} = updated_user} = Users.update_user(user, %{enabled: false})
+      assert updated_user.locked_at == locked_at
+    end
+
+    test "update_user/2 does unlock the user if enable true is passed as attribute and the user is locked" do
+      user = create_user()
+      {:ok, user} = Users.get_user(user.id)
+      assert {:ok, %User{} = user} = Users.update_user(user, %{enabled: false})
+
+      assert {:ok, %User{} = updated_user} = Users.update_user(user, %{enabled: true})
+      assert updated_user.locked_at == nil
+    end
+
+    test "update_user/2 does not update user with id 1" do
+      assert {:error, :operation_not_permitted} =
+               Users.update_user(%User{id: 1}, %{fullname: "new fullname"})
+    end
+
+    test "delete_user/2 does not delete user with id 1" do
+      assert {:error, :operation_not_permitted} = Users.delete_user(%User{id: 1})
+    end
+
+    test "delete_user/1 deletes the user" do
+      %{id: user_id, username: original_username} = user = create_user()
+      assert {:ok, %User{}} = Users.delete_user(user)
+      assert {:error, :not_found} == Users.get_user(user.id)
+
+      %User{deleted_at: deleted_at, username: username} = Trento.Repo.get_by!(User, id: user_id)
+
+      refute deleted_at == nil
+      refute username == original_username
+    end
+  end
+
+  # We can't use factory since we have to deal with pow logic
+  defp create_user do
+    password = "themightypassword8897"
+
+    {:ok, user} =
+      Users.create_user(%{
```

Comment:
Cannot we use a factory here?
You are implementing one at least hehe
I think it would be appropriate

Reply:
added factory!